packages/flutter_tools/lib/src/isolated/resident_web_runner.dart

```diff
@@ -335,7 +337,11 @@ Please provide a valid TCP port (an integer between 0 and 65535, inclusive).
           mainPath: target,
           debuggingOptions: debuggingOptions,
           platformArgs: <String, Object>{
-            'uri': url.toString(),
+             'uri': (debuggingOptions.tlsCertPath != null &&
+                         debuggingOptions.tlsCertKeyPath != null
+                     ? url.replace(scheme: 'https')
+                     : url)
+                 .toString(),
```

Comment:
Please, move this manipulation of the `url` to right next where it's created (Line ~305).

Also maybe check that both tlsCertPath and tlsCertKeyPath are different than null (or equal null) at the same time?

Reply:
Something like this?